nexus-and-alo/usage-scenarios.md

```diff
@@ -0,0 +1,32 @@
+
+**Contents:**
+
+- [Exposing Temporal application to a non-Temporal consumer (Nexus + ALO)](#exposing-temporal-application-to-a-non-temporal-consumer-nexus--alo)
+- [Exposing Temporal application to a Temporal consumer (Nexus)](#exposing-temporal-application-to-a-temporal-consumer-nexus)
+- [Exposing non-Temporal application with Temporal-esque calling semantics (ALO)](#exposing-non-temporal-application-with-temporal-esque-calling-semantics-alo)
+- [Exposing non-Temporal application via Nexus](#exposing-non-temporal-application-via-nexus)
+
+## Exposing Temporal application to a non-Temporal consumer (Nexus + ALO)
+
+**Problem statement:** One of the most common patterns we see from Temporal users is the need to expose a generic API for internal consumers who are not familiar with Temporal. The most common solution for this problem today is to stand up a separate infrastructure dependency in front of Temporal (such as a load balancer, HTTP server, GRPC Gateway) which defines the formal API & contract for consumers.
+
+**Solution:** Temporal users can now define and create a Nexus which represents the desired shape of their GRPC/HTTP API. Because Nexus gives users full control of how to parse/handle incoming requests, they can choose to expose their Temporal logic without anyone knowing that they used Temporal behind the scenes. If the consumer of their API has adopted the ALO standard, they can also “automagically” expose their Nexus APIs as ALOs. The capabilities of Nexus completely and totally obviate the need for a standalone API in these scenarios. 
+
+## Exposing Temporal application to a Temporal consumer (Nexus)
+
+**Problem statement:** Having a strong story for cross team collaboration is one of the most important aspects in the long term success of any technology. Within Temporal, we support the concept of a Namespace, which represents a bucket of isolated Temporal resources. While Namespaces are great for isolating resources, they leave a lot to be desired when it comes to exposing those resources to other teams operating inside or outside of your Temporal Cluster. Today cross Namespace calls only work with ChildWorkflows and even that behavior is unintuitive to most. Furthermore, calling into another Namespace when that Namespace resides on an external Cluster or even when it is replicated across your own Clusters makes things even less straightforward. 
+
+**Solution:** Temporal users can now define and create a Nexus which can be used to surface any Temporal functionality to other Temporal consumers. When calling into another Nexus from a Temporal context (eg: Workflow), every call will surfaced as an ALO, even if the Nexus implementation did not define them as such. This is because Temporal can automatically retry calls to any external endpoint and therefore seamlessly provide the ALO experience even when the endpoint was not implemented as an ALO originally. 
```

Comment:
This probably can't be entirely true. The automatic retrying will likely only meet user expectations if the endpoint is structured properly by providing idempotency.

It also raises the question, why would Nexus expose something that isn't an ALO? It's not clear to me at this point, reading serially, how an ALO and not-ALO are distinguished.

Reply:
> why would Nexus expose something that isn't an ALO? It's not clear to me at this point, reading serially, how an ALO and not-ALO are distinguished.

This tripped me up at first too, since technically "arbitrary" in ALO means it could be short. However, it does make sense that a user know whether the call is a request/response or a request/interactable-and-come-back-later. Basically all HTTP/gRPC calls today are the former, so the whole point of ALOs is to define the latter while letting Nexus support both.

Reply:
Chad said it well.